instrumentation/net/http/otelhttp/internal/semconv/v1.20.0_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package semconv
+
+import (
+	"testing"
+
+	"go.opentelemetry.io/otel/attribute"
+)
+
+func TestV120TraceRequest(t *testing.T) {
+	// Anything but "http" or "http/dup" works
+	t.Setenv("OTEL_HTTP_CLIENT_COMPATIBILITY_MODE", "old")
+	serv := NewHTTPServer()
+	want := func(req testServerReq) []attribute.KeyValue {
+		return []attribute.KeyValue{
+			attribute.String("http.method", "GET"),
+			attribute.String("http.scheme", "http"),
+			attribute.String("net.host.name", req.hostname),
+			attribute.Int("net.host.port", req.serverPort),
+			attribute.String("net.sock.peer.addr", req.peerAddr),
+			attribute.Int("net.sock.peer.port", req.peerPort),
+			attribute.String("user_agent.original", "Go-http-client/1.1"),
+			attribute.String("http.client_ip", req.clientIP),
+			attribute.String("net.protocol.version", "1.1"),
+			attribute.String("http.target", "/"),
+		}
+	}
+	testTraceRequest(t, serv, want)
+}
+
+func TestV120TraceResponse(t *testing.T) {
+	// Anything but "http" or "http/dup" works
+	t.Setenv("OTEL_HTTP_CLIENT_COMPATIBILITY_MODE", "old")
+	serv := NewHTTPServer()
+	want := []attribute.KeyValue{
+		attribute.Int("http.request_content_length", 701),
+		attribute.String("http.read_error", "read error"),
+		attribute.Int("http.response_content_length", 802),
+		attribute.String("http.write_error", "write error"),
+		attribute.Int("http.status_code", 200),
```

Comment:
These all seem like magic values. Why not structure this similar to the request test where they are passed from the testing harness?

Reply:
Because only the status code is contained within a response, the other fields are created from the package that use `semconv`.  This is to test that this package produces the Correct attributes from that data.

So all the data is made up by the test harness.

Reply:
> So all the data is made up by the test harness.

Right, but by magically referencing that data from this test it means any changes there will break here.

The `testTraceRequest` function takes a `func(req testServerReq) []attribute.KeyValue`. Why not have `testTraceResponse` accept a `func(req testServerResp) []attribute.KeyValue` argument and have things like the request/response length and status passed to it?